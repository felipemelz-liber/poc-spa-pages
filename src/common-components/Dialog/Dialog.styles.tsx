import styled from 'styled-components';

export const DialogHeader = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 24px;
`;

export const DialogActions = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-end;
  padding: 16px 24px;
  padding-top: 0px;
`;

export const DialogContent = styled.div`
  padding: 24px;
  padding-top: 0px;
  max-width: ${({ maxWidth }) => maxWidth || '522'}px;
`;

export const Title = styled.span`
  font-family: Roboto;
  font-size: 21px;
  font-weight: 500;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.4;
  letter-spacing: 0.08px;
  text-align: left;
  color: #4b6f85;
`;
