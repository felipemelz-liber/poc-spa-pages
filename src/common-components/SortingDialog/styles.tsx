import styled from 'styled-components';

export const SectionText = styled.span`
  font-family: Roboto;
  font-size: 16px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.38;
  letter-spacing: 0.01px;
  text-align: left;
  color: rgba(35, 39, 47, 0.87);
  margin-bottom: 16px;
`;

export const Column = styled.div`
  display: flex;
  flex-direction: column;
`;

export const Line = styled.div`
  display: flex;
  width: 100%;
  align-items: center;
  margin-bottom: 16px;
`;

export const Wrap = styled(Line)`
  min-width: 332px;
  margin-bottom: 32px;
  flex-wrap: wrap;
`;
