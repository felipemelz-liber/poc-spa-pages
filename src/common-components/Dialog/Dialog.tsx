import React from 'react';
import PropTypes from 'prop-types';
import { Modal, FlatButton } from 'liber-components';
import { DialogHeader, Title, DialogContent, DialogActions } from './Dialog.styles';
import LoadingButton from '../Buttons/LoadingButton';

const Dialog = ({ open, onClose, confirmationButtonProps, title, children, cancelLabel }) => {
  const { label, onConfirm, color, disabled, loading } = confirmationButtonProps;
  return (
    <Modal show={open} onLeaved={onClose}>
      <DialogHeader>
        <Title>{title}</Title>
      </DialogHeader>
      <DialogContent>{children}</DialogContent>
      <DialogActions>
        <FlatButton onClick={onClose}>{cancelLabel}</FlatButton>
        <LoadingButton
          ButtonComponent={FlatButton}
          color={color}
          onClick={onConfirm}
          disabled={disabled}
          loading={loading}
          spinnerColor={color === 'primary' ? '#009dff' : '#da2d16'}
          size="small"
        >
          {label || 'CONTINUAR'}
        </LoadingButton>
      </DialogActions>
    </Modal>
  );
};

Dialog.propTypes = {
  open: PropTypes.bool,
  onClose: PropTypes.func,
  confirmationButtonProps: PropTypes.shape({
    onConfirm: PropTypes.func,
    label: PropTypes.string,
    color: PropTypes.oneOf(['primary', 'danger']),
    disabled: PropTypes.bool,
    loading: PropTypes.bool,
  }),
  title: PropTypes.string,
  children: PropTypes.node,
  cancelLabel: PropTypes.string,
};

Dialog.defaultProps = {
  open: false,
  onClose: () => {},
  confirmationButtonProps: {
    onConfirm: () => {},
    label: 'CONTINUAR',
    color: 'primary',
    disabled: false,
    loading: false,
  },
  title: '',
  children: null,
  cancelLabel: 'CANCELAR',
};

export default Dialog;
