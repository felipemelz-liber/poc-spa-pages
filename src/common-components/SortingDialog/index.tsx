import React, { useState } from 'react';
import PropTypes from 'prop-types';

import { SectionText, Column, Wrap, Line } from './styles';
import Dialog from '../Dialog/Dialog';
import SelectTag from '../SelectTag/SelectTag';

const directions = [
  { label: 'Ascendente', value: 'asc' },
  { label: 'Descendente', value: 'desc' },
];

export const SortingDialog = ({ open, onClose, sorting, setSorting, columns }) => {
  const [selected, setSelected] = useState(null);
  const [direction, setDirection] = useState(null);
  const [disabled, setDisabled] = useState(false);

  React.useEffect(() => {
    const [sortedColumn = {}] = sorting.filter(({ columnName }) =>
      columns.map(({ value }) => value).includes(columnName),
    );
    const { columnName, direction: selectedDirection } = sortedColumn;

    setSelected(columnName);
    setDirection(selectedDirection);
  }, [sorting]);

  React.useEffect(() => {
    setDisabled(!selected || !direction);
  }, [selected, direction]);

  const onConfirm = () => {
    setSorting({ columnName: selected, direction });
    onClose();
  };

  return (
    <Dialog
      open={open}
      onClose={onClose}
      title="Ordenar Fornecedor"
      confirmationButtonProps={{ label: 'ORDENAR', onConfirm, disabled }}
    >
      <Column>
        <SectionText>Ordenar por</SectionText>
        <Wrap>
          {columns.map(({ label, value }) => (
            <SelectTag
              key={value}
              selected={selected === value}
              onChange={() => setSelected(value)}
            >
              {label}
            </SelectTag>
          ))}
        </Wrap>
        <SectionText>Ordem</SectionText>
        <Line>
          {directions.map(({ label, value }) => (
            <SelectTag
              key={value}
              selected={direction === value}
              onChange={() => setDirection(value)}
            >
              {label}
            </SelectTag>
          ))}
        </Line>
      </Column>
    </Dialog>
  );
};

SortingDialog.propTypes = {
  open: PropTypes.bool,
  onClose: PropTypes.func,
  sorting: PropTypes.arrayOf(PropTypes.object),
  setSorting: PropTypes.func,
  columns: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string,
      value: PropTypes.string,
    }),
  ),
};

SortingDialog.defaultProps = {
  open: false,
  onClose: () => null,
  sorting: [],
  setSorting: () => null,
  columns: [],
};

export default SortingDialog;
