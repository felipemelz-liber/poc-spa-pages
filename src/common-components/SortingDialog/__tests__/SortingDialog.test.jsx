import React from 'react';
import sinon from 'sinon';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import SortingDialog from '..';
import Dialog from '../../Dialog/Dialog';

describe('SortingDialog tests', () => {
  configure({ adapter: new Adapter() });
  const onClose = sinon.spy();
  const setSorting = sinon.spy();
  const props = {
    onClose,
    setSorting,
    columns: [
      {
        label: 'Column One',
        value: 'col_1',
      },
      {
        label: 'Column Two',
        value: 'col_2',
      },
    ],
  };

  afterEach(() => {
    sinon.resetHistory();
  });

  it('should render correctly without props', () => {
    const wrapper = shallow(<SortingDialog />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call changeSorting on confirm', () => {
    const wrapper = shallow(<SortingDialog {...props} />);
    const confProps = wrapper.find(Dialog).prop('confirmationButtonProps');
    confProps.onConfirm();

    expect(setSorting.called).toBeTruthy();
  });

  it('should call onClose on dialog close', () => {
    const wrapper = shallow(<SortingDialog {...props} />);
    wrapper.find(Dialog).simulate('close');

    expect(onClose.called).toBeTruthy();
  });
});
