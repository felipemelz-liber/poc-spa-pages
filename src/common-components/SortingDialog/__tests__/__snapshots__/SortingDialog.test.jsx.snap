// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortingDialog tests should render correctly without props 1`] = `
<Dialog
  cancelLabel="CANCELAR"
  confirmationButtonProps={
    Object {
      "disabled": false,
      "label": "ORDENAR",
      "onConfirm": [Function],
    }
  }
  onClose={[Function]}
  open={false}
  title="Ordenar Fornecedor"
>
  <styled.div>
    <styled.span>
      Ordenar por
    </styled.span>
    <Styled(styled.div) />
    <styled.span>
      Ordem
    </styled.span>
    <styled.div>
      <SelectTag
        key="asc"
        onChange={[Function]}
        selected={false}
      >
        Ascendente
      </SelectTag>
      <SelectTag
        key="desc"
        onChange={[Function]}
        selected={false}
      >
        Descendente
      </SelectTag>
    </styled.div>
  </styled.div>
</Dialog>
`;
